libp2p/services/autonatservice.nim

```diff
@@ -0,0 +1,127 @@
+# Nim-LibP2P
+# Copyright (c) 2022 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import std/[tables]
+import chronos
+import ../switch
+import ../protocols/[connectivity/autonat,
+                    rendezvous]
+import ../protocols/connectivity/relay/[relay, client]
+import ../discovery/[rendezvousinterface, discoverymngr]
+import ../utils/heartbeat
+import ../crypto/crypto
+
+type
+  AutonatService* = ref object of Service
+    registerLoop: Future[void]
+    scheduleInterval: Duration
+    networkReachability: NetworkReachability
+    t: CountTable[NetworkReachability]
+    autonat: Autonat
+    newStatusHandler: NewStatusHandler
+    rng: ref HmacDrbgContext
+    numPeersToAsk: int
+    maxConfidence: int
```

Comment:
one last comment, not sure if nitpick. `minConfidence` is a better naming? As i see it is the minimum amount of states that we use to consider that the node is `Reachable` or `NotReachable`